cirq-core/cirq/ops/state_preparation_gate.py

```diff
@@ -0,0 +1,125 @@
+# Copyright 2018 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Quantum gates defined by a matrix."""
+
+from typing import Any, Sequence, Dict, List, Tuple, TYPE_CHECKING
+
+import numpy as np
+
+from cirq import protocols
+from cirq.ops import raw_types
+from cirq.ops.common_channels import ResetChannel
+from cirq.ops.matrix_gates import MatrixGate
+from cirq._compat import proper_repr
+
+if TYPE_CHECKING:
+    import cirq
+
+
+class PrepareState(raw_types.Gate):
```

Comment:
There's a pretty strong precedent in Cirq for naming gates with the format `*Gate`, and using shorter names for specific implementations of those gates or protocols that produce them.

In keeping with this precedent (and based on the default name), the gate should instead be named `StatePreparationGate`.